Remove v1 content building support

athos-ribeiro opened the pull request:
* OSBS-6665

https://github.com/projectatomic/osbs-docs/pull/76

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates

twaugh commented on tests/plugins/test_orchestrate_build.py:
```diff
@@ -1144,80 +1144,6 @@ def test_orchestrate_override_build_kwarg(tmpdir, overrides):
-    build_result = runner.run()
-    if will_fail:
-        assert build_result.is_failed()
-        assert 'failed to create worker build' in caplog.text
```
We seem to be missing some coverage now: https://coveralls.io/builds/23631116/source?filename=atomic_reactor%2Fplugins%2Fbuild_orchestrate_build.py#L621

athos-ribeiro replied:
I changed the `_update_content_versions` method (instead of removing it) to validate the contents in the `content_versions` list.

athos-ribeiro commented:
Release notes added:

# Improvements

- Dropped support for container registry API V1
- Dropped support for V1 content

I also removed the deprecation warning note for API V1
